### DISCUSSION TITLE
Aberrant Splicing implementation:PCA not autoencoder?

### DISCUSSION THREAD

**Discussion by Nico282828** [gagneurlab/drop#530] [Q&A]
Dear DROP community,

  I hope all of you are doing well! I am currently new to using this pipeline and have realized that for AE, autoencoder is an option for the correction of covariations between samples. But for AS, there are only two options "PCA" and "PCA-BB-Decoder". 

  From the DROP paper, it mentioned that both OUTRIDER and FRASER are able to perform autoencoder by fitting a beta binomal distribution to correct for sample covariates. I wonder if we are able to choose "autoencoder" instead for AS. 

  Or please do enlighten me if I am misunderstanding the meaning of "implementation" option in both modules. 

  Thanks a lot! 

Best Wishes,
Nico

**Comment by ischeller** [comment]
Hi @Nico282828 ,
for the AS pipeline, "PCA" and "PCA-BB-Decoder" are the two options we recommend using, mostly due to runtime considerations. However, it's possible to run the full autoencoder implementation for FRASER inside DROP as well by specifying "AE" in the config field for implementation. Be aware, however, that running the full autoencoder fit in FRASER will significantly increase the runtime and will take significantly longer than the autoencoder fit of OUTRIDER as there are typically ~10x more introns than genes.  It did also not really improve the performance compared to the options using PCA in our benchmarks which is why we recommend using one of "PCA" or "PCA-BB-Decoder" for most users.

From the help page of the main FRASER function (`?FRASER`): 
"Available methods to correct for the confounders are currently: a denoising autoencoder with a BB loss ("AE" and "AE-weighted"), PCA ("PCA"), a hybrid approach where PCA is used to fit the latent space and then the decoder of the autoencoder is fit using the BB loss ("PCA-BB-Decoder")."

Hope this helps!

Best,
Ines

**Comment by Nico282828** [comment]
Dear Ines,

  Thanks a lot for your help! 

Best Wishes,
Nico
